src/ReservoirPair.sol

```diff
@@ -148,15 +150,47 @@ abstract contract ReservoirPair is IAssetManagedPair, ReservoirERC20 {
         require(aBalance0 <= type(uint104).max && aBalance1 <= type(uint104).max, "RP: OVERFLOW");
         require(aReserve0 <= type(uint104).max && aReserve1 <= type(uint104).max, "RP: OVERFLOW");
 
-        uint32 lBlockTimestamp = uint32(_currentTime());
+        uint32 lBlockTimestamp = _currentTime();
         uint32 lTimeElapsed;
         unchecked {
             // underflow is desired
             // however in the case where no swaps happen in ~68 years (2 ** 31 seconds) the timeElapsed would underflow twice
             lTimeElapsed = lBlockTimestamp - aBlockTimestampLast;
-        }
-        if (lTimeElapsed > 0 && aReserve0 != 0 && aReserve1 != 0) {
-            _updateOracle(aReserve0, aReserve1, lTimeElapsed, lBlockTimestamp);
+
+            // both balance should never be zero, but necessary to check so we don't pass 0 values into arithmetic operations
+            // shortcut to calculate aBalance0 > 0 && aBalance1 > 0
+            if (aBalance0 * aBalance1 > 0) {
+                Observation storage lPrevious = _observations[sSlot0.index];
+                (uint256 lInstantRawPrice, int256 lLogInstantRawPrice) = _calcSpotAndLogPrice(aBalance0, aBalance1);
+
+                // a new sample is not written for the first mint
+                // shortcut to calculate lTimeElapsed > 0 && aReserve0 > 0 && aReserve1 > 0
+                if (lTimeElapsed * aReserve0 * aReserve1 > 0) {
```

Comment:
just want to document this for posterity: 

I attempted to simplify this condition to only `lTimeElapsed > 0`, which therefore implies that an oracle sample is created in the very first mint. However, I did not adopt this approach in the end for one edge case. In the edge case of a swap happening after the very first mint (pair creation). the previous timestamp would not be zero, and therefore the clamped price would not be correctly calculated (it'd always be 0 cuz the time elapsed is 0). 

So it's back to the original approach of writing only the instant prices for the very first block (regardless of how many things happen in that block), and use the previous oracle observation as a "placeholder" to temporarily store the instant prices for accumulation in the future. This approach is more semantically correct as we're not "creating" a sample (as we don't write a timestamp), but instead just using it as a placeholder.